src/main/java/com/neprozorro/service/LotInfoService.java

```diff
@@ -93,18 +93,21 @@ private Predicate inOrEqualLotStatus(Root<LotInfo> root, CriteriaBuilder criteri
         return root.get(column).in(lotStatuses);
     }
 
-    private Predicate betweenOrEqual(Root<LotInfo> root, CriteriaBuilder criteriaBuilder, String column, String value) {
+    private Predicate rangeTotalPrice(Root<LotInfo> root, CriteriaBuilder criteriaBuilder, String column, String value) {
         String[] parsedValue = value.split(",");
 
-        if (parsedValue.length == 1) {
-            return criteriaBuilder.equal(root.get(column), BigDecimal.valueOf(Double.valueOf(value)));
-        }
+        BigDecimal minValue = BigDecimal.valueOf(Double.parseDouble(parsedValue[0]));
+        BigDecimal maxValue = BigDecimal.valueOf(Double.parseDouble(parsedValue[1]));
+
+        if (parsedValue.length != 2 ) throw new RuntimeException("Invalid input format");
 
-        if (parsedValue.length != 2) {
-            throw new RuntimeException();
+        if (maxValue.compareTo(BigDecimal.ZERO) == 0 || maxValue == null) {
+            return criteriaBuilder.greaterThan(root.get(column), minValue);
         }
 
-        return criteriaBuilder.between(root.get(column), BigDecimal.valueOf(Double.valueOf(parsedValue[0])),
-                BigDecimal.valueOf(Double.valueOf(parsedValue[1])));
+        int comparing = minValue.compareTo(maxValue);
+        if (comparing > 0) throw new RuntimeException("k;ugilugiugilu");
```

Comment:
The same as for the previous comment. Additionally wrong message and check your PR code when it's on Github after you checked it during the push. It should be minimum doublecheck and better to check it 3 times.

Reply:
done